app/src/main/java/ch/epfl/sdp/blindly/main_screen/match/MatchPageFragment.kt

```diff
@@ -232,16 +247,17 @@ class MatchPageFragment : Fragment(), CardStackListener {
         if (users == null) {
             return listOf()
         }
-        val userid = userHelper.getUserId()!!
-        val currentUser = userRepository.getUser(userid)
+        currentUserId = userHelper.getUserId()!!
+        currentUser = userRepository.getUser(currentUserId)!!
```

Comment:
I think if you were to use a viewModel here instead of calling directly the repository in the UI that you'd be able to maybe test the match fragment a bit more